dags/remote_automated_ampliseq.py

```diff
@@ -0,0 +1,373 @@
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def transfer_compressed_results(study):
+            logging.info(f"Transferring compressed results for study {study['study']}...")
+            remote_compressed_path = os.path.join(REMOTE_PATH, f"{study['study']}_results.tar.gz")
+            local_compressed_path = os.path.join(study['path'], f"{study['study']}_results.tar.gz")
+            sftp_op = SFTPOperator(
+                task_id='transfer_compressed_results_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SFTP,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                local_filepath=local_compressed_path,
+                remote_filepath=remote_compressed_path,
+                operation='get',
+                create_intermediate_dirs=True
+            )
+            sftp_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def extract_results_file(study):
+            logging.info(f"Extracting results file for study {study['study']}...")
+            local_compressed_path = os.path.join(study['path'], f"{study['study']}_results.tar.gz")
+            local_extract_path = study['path']
+            with tarfile.open(local_compressed_path, 'r:gz') as tar:
+                tar.extractall(path=local_extract_path)
+            os.remove(local_compressed_path)
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def local_Rscript_postprocessing(study):
+            logging.info(f"Running local Rscript postprocessing for study {study['study']}...")
+            command = f'Rscript {BASE_PATH}/ampliseq_postProcessing.R {study["study"]} {study["path"]}/out'
+            os.system(command)
+            logging.info(f"Postprocessing done for study {study['study']}.")
+
+        @dag.task()
+        def clean_up_remote():
+            logging.info("Cleaning up remote directory...")
+            command = f'rm -rf {REMOTE_PATH}/*'
+            ssh_op = CustomSSHOperator(
+                task_id='clean_up_remote',
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task()
+        def clean_up_local():
+            clean_up_command = f"find {BASE_PATH}/* -type f \\( -name '*.tar.gz' -o -name '*.sh' \\) -exec rm {{}} +"
+            return BashOperator(
+                task_id='clean_up_local',
+                bash_command=clean_up_command,
+            ).execute(context={})
+
+        loaded_studies = load_studies()
+
+        with TaskGroup("process_amplicon_studies") as process_amplicon_studies:
+            script_names_nextflow = create_shell_script_nextflow.expand(study=loaded_studies)
```

Comment:
so this is interesting. it looks like rather than attempt to expand on a task group, youve expanded on every task within the task group. that probably is avoiding some of the frustrating issues i was seeing, but is also going to probably produce a very confusing looking graph..

is there a reason (until airflow decides to better support using dynamic task mapping, task groups and branching operators together) to not build the dag manually in a loop? i know you were doing exactly that before and i was the one suggested trying something different.. but thats clearly been more trouble than its worth. particularly if its also caused you to split this dag in two for remote vs local..